lambda_function.py

```diff
@@ -59,20 +41,25 @@ def get_collection_schedule(event) -> tuple:
     query_gis['geometry'] = f'{long},{lat}'
     gis_r: dict = c_session.get(GIS_URL, params=query_gis).json()
     area_date = gis_r['features'][0]['attributes']['AREA_NAME'].replace(' ', '')
-    end_week = get_end_of_week(area_date[:-1])
+    end_week = datetime.datetime.now().date().isocalendar()
+    index = 1
     gsheet = c_session.get(SHEET_URL).text
     csv_dict = csv.DictReader(StringIO(gsheet))
     for row in csv_dict:
-        if row["Calendar"] == area_date and row["WeekStarting"] == end_week:
-            return row, csv_dict.__next__()
+        if row["Calendar"] == area_date:
+            if row["Calendar"] == area_date and end_week[1] == index and int(row['WeekStarting'][:4]) == end_week[0]:
+                return row, csv_dict.__next__()
+            elif row["Calendar"] == area_date and row['WeekStarting'][:4] == end_week[0]:
```

Comment:
These inner `if` and `elif` statements first conditional seems unnecessary as it these statements wouldn't even be run if `row["Calender"` was not equal to `area_date`.

Reply:
Thanks @42B. fixed.